### DISCUSSION TITLE
Admin 'override' for insert options

### DISCUSSION THREAD

**Discussion by Luuk1983** [umbraco/Umbraco-CMS#17759] [Features and ideas]
Sometimes you just need to have a node of a certain type one time and you don't want anyone else to create it. Lets take a 404 page below a homepage node as an example.

So now I need to add the 404 document type to the homepage as a valid insert option, create tot 404 page in content and then remove the 404 page from the homepage again. In Sitecore (ew, dirty word!), you can ALWAYS override the insert options and practically create any document type anywhere in the tree.

![afbeelding](https://github.com/user-attachments/assets/1047a86b-d7e8-4936-90c3-98403b2ffd96)
(Yes, a 'template' in Sitecore is a content type in Umbraco)

And although I don't miss Sitecore at all, this is a small feature that's really nice to have as an administrator.

**Comment by skttl** [comment]
The API (at least in v13 and before) doesn't prevent you from adding nodes of disallowed content types where you need it. So you might be able to make a simple extension for Umbraco, that could add a menu item for "admin create node".

If you notice the url when creating a new node, you can see the different parts that make it up:
`/umbraco#/content/content/edit/1462?mculture=fr-BE&cculture=fr-BE&doctype=contentPage&create=true`

Notice the doctype parameter. You can put in whatever content type alias you want to create, and nothing prevents you from doing that.

So if you can solve the UI part in an extension, you've got yourself a nifty little package, that other people might find handy :)

**Comment by skttl** [comment]
Just tried in v15 - and you while the url seems to allow for the same manipulation (although with guids instead of aliases), the API now seems to prevent you from creating disallowed content types.

**Reply by Luuk1983** [reply]
Ow that's too bad, was about to make a simpel package for it! But this is a bummer. I seem to remember (but not entirely sure) that uSync also doesn't like it very much when it needs to add content that is technically not allowed to be there.

**Comment by dawoe** [comment]
@Luuk1983 

In v13 you could use create a notificationhandler for this : https://docs.umbraco.com/umbraco-cms/13.latest/reference/notifications/sendingallowedchildrennotifications

This is still found in the docs for V14 and V15 but if I am correct this has been removed from Umbraco. Don't know what the alternative is in the new backoffice

**Reply by rsoeteman** [reply]
I really hope this will be documented soon otherwise a lot of implemenations can't be upgraded to V17 :-(

**Comment by rsoeteman** [comment]
@Luuk1983 

> Sometimes you just need to have a node of a certain type one time and you don't want anyone else to create it. Lets take a 404 page below a homepage node as an example.
> 
> So now I need to add the 404 document type to the homepage as a valid insert option, create tot 404 page in content and then remove the 404 page from the homepage again. In Sitecore (ew, dirty word!), you can ALWAYS override the insert options and practically create any document type anywhere in the tree.
> 
> ![afbeelding](https://private-user-images.githubusercontent.com/9865589/393785906-1047a86b-d7e8-4936-90c3-98403b2ffd96.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM4MjA5NDUsIm5iZiI6MTczMzgyMDY0NSwicGF0aCI6Ii85ODY1NTg5LzM5Mzc4NTkwNi0xMDQ3YTg2Yi1kN2U4LTQ5MzYtOTBjMy05ODQwM2IyZmZkOTYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIxMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMTBUMDg1MDQ1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzdiYjU1YzZhZDI3ZTRiNmVkOGYxMThjYjc1YzU1NjMzMjk0Y2I1YmU4MTU4YmU2MGVhODkzYzExNjE1MjA5NSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.jxF1efvnq7ANj4F3tx-RvIAEYAzKlDR9X6TlaId656c) (Yes, a 'template' in Sitecore is a content type in Umbraco)
> 
> And although I don't miss Sitecore at all, this is a small feature that's really nice to have as an administrator.

Check the presentation I left you with earlier this year for samples ;-)

**Comment by Luuk1983** [comment]
@rsoeteman I've made extensive use of your example, we have implemented easy ways to restrict nodes to an x number of child nodes of a certain type below a parent and disallow delete of certain nodes, unless you are an admin (and I changed the menu option so it's clear when you are an admin that you aren't really supposed to delete the node).

Creating an additional menu to override insert options (I should call it permissions in Umbraco) isn't that hard. But if it doesn't work in Umbraco14+, I'm not gonna bother. I also hope that there is an alternative to the SendingAllowedChildrenNotification, because I'm about to update our core packages to Umbraco 15. That would break all that functionality...

**Reply by rsoeteman** [reply]
Ah sorry, yes this must be possible in V14+ (Before next LTS releases) Otherwise it will prevent a lot of implementations from Upgrading to VNext Backoffice. Maybe @nielslyngsoe knows what to use in V14+?

**Comment by nul800sebastiaan** [comment]
Just to crosslink this back to the chat on  Discord around `SendingAllowedChildrenNotification`: https://discord-chats.umbraco.com/t/18797221/has-the-sendingallowedchildrennotification-been-deprecated-f

And the link if you're already on Discord and want to talk more there: https://discord.com/channels/869656431308189746/1238039395748020275
